boards/esp32-ethernet-kit/include/board.h

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (C) 2018 Gunar Schorcht
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     boards_esp32_esp-ethernet-kit
+ * @file
+ * @{
+ */
+
+#ifndef BOARD_H
+#define BOARD_H
+
+#include <stdint.h>
+
+/* include common board definitions as last step */
+#include "board_common.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Initialize the board specific hardware
+ */
+static inline void board_init(void) {
+    /* there is nothing special to initialize on this board */
+    board_init_common();
+}
+
+/**
```

Comment:
As long as module `esp_eth` isn't used, the `RESET_N` pin GPIO5 isn't configured as output. Since `RESET_N` is pulled down, the PHY oscillator is disabled. Thus, you should be able to use GPIO0 and the `BOOT` as `BTN0`.

```suggestion
#if MODULE_ESP_ETH || DOXGEN
/**
 * @name    Button pin definitions
 * @{
 */

/**
 * @brief   Default button GPIO pin definition
 */
#define BTN0_PIN        GPIO0

/**
 * @brief   Default button GPIO mode definition
 *
 * Since the GPIO of the button is pulled up with an external resistor, the
 * mode for the GPIO pin has to be GPIO_IN.
 */
#define BTN0_MODE       GPIO_IN

/**
 * @brief   Default interrupt flank definition for the button GPIO
 */
#ifndef BTN0_INT_FLANK
#define BTN0_INT_FLANK  GPIO_FALLING
#endif

/**
 * @brief   Definition for compatibility with previous versions
 */
#define BUTTON0_PIN     BTN0_PIN
/** @} */
#endif

/**
```

Reply:
Added this, but without ifdef around it - similar to comment about EMAC defines just below.

Reply:
Please use the `#ifdef ... #endif` since there are application that test for the existence of these definitions. If you don't have the button, you must not define them. It is different for EMAC defines.

Reply:
Done.

Reply:
Did you test whether it works with `tests/buttons`?